lib/common/verify-node-version.ts

```diff
@@ -83,13 +82,13 @@ export function getNodeWarning(): ISystemWarning {
 		}
 
 		if (warningMessage) {
-			return {
+			nodeWarn = {
 				message: warningMessage,
 				severity: SystemWarningsSeverity.medium
 			};
 		}
-
-		return null;
 	}
+
+	return nodeWarn;
```

Comment:
Maybe you can return a deep copy, so that consumers can't modify the warning. Not a merge blocker.

Reply:
I'm not sure about this, so I'll keep it this way for the moment and I'll keep it in mind in case such issue occurs in the future. Thanks for the suggestion.